### ISSUE TITLE
Question: ArangoRepository.save usage

### ISSUE THREAD

**Issue by ZachHerridge** [arangodb/spring-data#92]
We have documents which have fields annotated like:
`    
@Id
private String id;

@Key
private String key;
`

Previously setting the key field then calling ArangoRepository.save would result in the document being saved. With today's release if a document does not already exist with the key a error is thrown saying the document does not exist which is true. Is this how save should work?

**Comment by mpv1989** [comment]
Hi @ZachHerridge ,

we had to change the `ArangoOperations#upsert` method, which is used by `ArangoRepository.save()` because it didn't worked when one is only using the `@Key` annotation and not `@Id` in addition. Sadly we forgot the case when someone is using user generated keys, like you probably do.

I talked with a colleague about the correct Spring way for this case and we came up with the solution of Spring Data's `Persistable` interface. I pushed a fix and will release a bugfix release today.

The usage of `Persistable` is very simple but unfortunately also a bit ugly. Your entities have to implement the `Persistable` interface and you have to find a way that `isNew()` is returning `true` the first time you are saving your entity. You can simply do that with a boolean flag which you have to set manually after first saving the entity (don't forget to annotate it with `@Transient`).

```java
public static class ExampleEntity implements Persistable<String> {

  @Id private String id;
  @Key private final String key;
  @Transient private transient boolean persisted;

  public ExampleEntity(final String key) {
    super();
    this.key = key;
  }

  public void setPersisted(final boolean persisted) {
    this.persisted = persisted;
  }

  @Override
  public boolean isNew() {
    return !persisted;
  }

  @Override
  public String getId() {
    return key; // or `id`, does not matter
  }
}
```

All this becomes unnecessary as soon as ArangoDB 3.4 is out and you are able to upgrade to it. With ArangoDB 3.4 comes REPSERT (replace-insert) which is used from `ArangoRepository#save()`. Then you don't have to use `Persistable` anymore and everything should work as expected.

**Comment by mpv1989** [comment]
I released version 1.2.1 and 2.2.1 including the fix.

I will close this issue, feel free to open it again if something does not work as expected.
